### PR TITLE
Add moderator notes for editing replies

### PR DESCRIPTION
**Pull request by Marri**
Audits for moderators part one: editing replies

### CODE REVIEW

File: app/assets/javascripts/posts/editor.js [hunk 1]
```diff
@@ -12,9 +12,7 @@ $(document).ready(function() {
     formButtons.not(this).data('disable-with', '').prop('disabled', true);
   });
 
-  if ($("#post-editor .view-button").length === 0) return; // Skip if there is no writable editor (no RTF/HTML buttons)
-
-  setupWritableEditor();
+  if ($("#post-editor .view-button").length > 0) setupWritableEditor();
```
Comment (Marri) [inline app/assets/javascripts/posts/editor.js]: Unrelated, but stumbled across it; should be the same functionality, but less of a gotcha in future if someone adds code below the return.

File: app/controllers/replies_controller.rb [hunk 2]
```diff
@@ -166,8 +166,22 @@ def update
...
-    @reply.skip_post_update = true unless @reply.post.last_reply_id == @reply.id
-    @reply.save
+    if current_user.id != @reply.user_id && @reply.audit_comment.blank?
+      flash[:error] = "You must provide a reason for your moderator edit."
+      editor_setup
+      render action: :edit and return
+    end
+
+    @reply.audit_comment = nil if @reply.changes.empty? # don't save an audit for a note and no changes
+    @reply.skip_post_update = true unless @reply.post.try(:last_reply_id) == @reply.id
```
Comment (Marri) [inline app/controllers/replies_controller.rb]: Also not moderation relevant, but added the missing reply save validation check here, along with a test; since my test checks by setting post to `nil`, also needed a `try()`.

File: app/controllers/writable_controller.rb [hunk 3]
```diff
@@ -119,7 +120,7 @@ def display_warnings?
 
   def editor_setup
     use_javascript('posts/editor')
-    build_template_groups
+    build_template_groups(@reply.try(:user) || @post.try(:user))
```
Comment (Marri) [inline app/controllers/writable_controller.rb]: `@reply` must come first here because most reply controller methods set `@post` as well, and the user isn't necessarily the same.
